tests/ServerTest.php

```diff
@@ -1023,6 +1023,162 @@ function ($data) use (&$buffer) {
         $this->assertInstanceOf('InvalidArgumentException', $error);
     }
 
+    public function testInvalidChunkHeaderResultsInErrorResponse()
```

Comment:
The updated tests now look good to me and I agree that it makes sense to re-organize this in a later follow-up PR :+1: